mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -739,6 +745,9 @@ void mav_vehicle::takeoff()
 
 void mav_vehicle::rotate(double angle_deg, bool autocontinue)
 {
+    // We have taken control
+    this->is_our_control = true;
```

Comment:
It seems a bit weird that you set control to true here. It you are asking the drone to rotate, you should check if you have the control of the drone, not set it to true. Can't we identify that we got the control back from any of the mavlink messages we receive? 
Same applies for the other functions below.